tests/test_protocol.py

```diff
@@ -139,14 +132,7 @@ def do_test():
 
         done.append(True)
 
-    thread = threading.Thread(target=do_test)
-    thread.start()
-
-    while len(done) == 0 and limit > 0:
-        await gen.sleep(sleep)
-        limit -= 1
-
-    thread.join()
+    await asyncio.wait_for(asyncio.get_running_loop().run_in_executor(None, do_test), timeout=1)
```

Comment:
I think this is equivalent but please review this carefully.

Reply:
I agree